internal/server/server.go

```diff
@@ -217,15 +220,14 @@ func registerUIRoutes(router *gin.Engine, opts UIOptions) {
 
 func (s *Server) listen() error {
 	ginutil.SetMode()
-	promRegistry := setupMetrics(s.DB())
-	router := s.GenerateRoutes(promRegistry)
+	router := s.GenerateRoutes()
 
 	httpErrorLog := log.New(logging.NewFilteredHTTPLogger(), "", 0)
 	metricsServer := &http.Server{
 		ReadHeaderTimeout: 30 * time.Second,
 		ReadTimeout:       60 * time.Second,
 		Addr:              s.options.Addr.Metrics,
-		Handler:           metrics.NewHandler(promRegistry),
+		Handler:           metrics.NewHandler(s.metricsRegistry),
```

Comment:
Seems like we should handle the possibility of this value being nil in the metrics functions. Feels like a likely nil pointer de-reference.

Reply:
In what scenario would we expect the metrics registry to be nil?

In production we set this to a value in `server.New`, so it can't ever be nil. I think for this to be nil it would have to be a programming error, which should panic.